frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/BaseResetBranchAction.java

```diff
@@ -0,0 +1,169 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getGitMacheteRepository;
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getProject;
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getSelectedVcsRepository;
+
+import com.intellij.dvcs.DvcsUtil;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import com.intellij.openapi.actionSystem.Presentation;
+import com.intellij.openapi.application.AccessToken;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import git4idea.GitUtil;
+import git4idea.branch.GitBranchUiHandlerImpl;
+import git4idea.branch.GitBranchWorker;
+import git4idea.commands.Git;
+import git4idea.commands.GitCommand;
+import git4idea.commands.GitCommandResult;
+import git4idea.commands.GitLineHandler;
+import git4idea.repo.GitRepository;
+import git4idea.repo.GitRepositoryManager;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.BaseGitMacheteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRemoteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepository;
+import com.virtuslab.gitmachete.backend.api.SyncToRemoteStatus;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+import com.virtuslab.logger.IPrefixedLambdaLogger;
+import com.virtuslab.logger.PrefixedLambdaLoggerFactory;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_GIT_MACHETE_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public abstract class BaseResetBranchAction extends GitMacheteRepositoryReadyAction {
+  private static final IPrefixedLambdaLogger LOG = PrefixedLambdaLoggerFactory.getLogger("frontendActions");
+  private static final String VCS_LOGGER_TITLE = "Resetting";
+  private static final String TASK_TITLE = "Resetting...";
+
+  protected abstract Option<String> getBranchName(AnActionEvent anActionEvent);
+
+  protected abstract String getActionType();
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    Presentation presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    Option<String> branchName = getBranchName(anActionEvent);
+
+    if (branchName.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription("Reset disabled due to undefined ${getActionType()} branch");
+      return;
+    }
+
+    Option<SyncToRemoteStatus> syncToRemoteStatus = getGitMacheteRepository(anActionEvent)
```

Comment:
Use helper for branch name

Reply:
Can you explain what do you mean?
I have branch name in var, so I just use it.

Reply:
Ok, leave as is, I confused two things here